packages/react-next/src/components/OverflowSet/OverflowSet.base.tsx

```diff
@@ -1,257 +1,225 @@
-      focusSucceeded = this._focusZone.current.focus(forceIntoFirstElement);
-    }
-
-    return focusSucceeded;
-  }
-
-  /**
-   * Sets focus to a specific child element within the OverflowSet.
-   * @param childElement - The child element within the zone to focus.
-   * @returns True if focus could be set to an active element, false if no operation was taken.
-   */
-  public focusElement(childElement?: HTMLElement): boolean {
-    let focusSucceeded = false;
-
-    if (!childElement) {
-      return false;
-    }
-
+    focusZoneProps,
+    styles,
+    vertical,
     // eslint-disable-next-line deprecation/deprecation
-    if (this.props.doNotContainWithinFocusZone) {
-      if (this._divContainer.current && elementContains(this._divContainer.current, childElement)) {
-        childElement.focus();
-        focusSucceeded = document.activeElement === childElement;
-      }
-    } else if (this._focusZone.current) {
-      focusSucceeded = this._focusZone.current.focusElement(childElement);
-    }
+    doNotContainWithinFocusZone,
+    role,
+    overflowSide = 'end',
+  } = props;
+  let Tag;
+  let uniqueComponentProps;
 
-    return focusSucceeded;
-  }
+  const classNames: IProcessedStyleSet<IOverflowSetStyles> = getClassNames(styles, { className, vertical });
 
-  // Add keytip register/unregister handlers to lifecycle functions to correctly manage persisted keytips
-  public componentDidMount() {
-    this._registerPersistedKeytips();
-  }
-
-  public componentWillUnmount() {
-    this._unregisterPersistedKeytips();
-  }
+  const showOverflow = overflowItems && overflowItems.length > 0;
 
-  public UNSAFE_componentWillUpdate() {
-    this._unregisterPersistedKeytips();
-  }
-
-  public componentDidUpdate() {
-    this._registerPersistedKeytips();
-  }
-
-  private _registerPersistedKeytips() {
-    Object.keys(this._persistedKeytips).forEach((key: string) => {
-      const keytip = this._persistedKeytips[key];
-      const uniqueID = this._keytipManager.register(keytip, true);
-      // Update map
-      this._persistedKeytips[uniqueID] = keytip;
-      delete this._persistedKeytips[key];
-    });
-  }
-
-  private _unregisterPersistedKeytips() {
-    // Delete all persisted keytips saved
-    Object.keys(this._persistedKeytips).forEach((uniqueID: string) => {
-      this._keytipManager.unregister(this._persistedKeytips[uniqueID], uniqueID, true);
-    });
-    this._persistedKeytips = {};
-  }
-
-  private _onRenderItems = (items: IOverflowSetItemProps[]): JSX.Element[] => {
-    return items.map((item, i) => {
-      return (
-        <div key={item.key} className={this._classNames.item}>
-          {this.props.onRenderItem(item)}
-        </div>
-      );
-    });
-  };
-
-  private _onRenderOverflowButtonWrapper = (items?: IOverflowSetItemProps[]): JSX.Element => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const onRenderOverflowButtonWrapper = (itemsProp: any[]): JSX.Element => {
```

Comment:
Can we use the proper type `IOverflowSetItemProps` here (and a couple lines down)?